packages/mask/src/plugins/Trader/trader/1inch/useTradeGasLimit.ts

```diff
@@ -0,0 +1,27 @@
+import type { SwapQuoteOneResponse, TradeComputed } from '../../types'
+import { useMemo } from 'react'
+import { useAccount, useWeb3 } from '@masknet/web3-shared-evm'
+import { TargetChainIdContext } from '../useTargetChainIdContext'
+import { pick } from 'lodash-unified'
+import type { TransactionConfig } from 'web3-core'
+import { useAsync } from 'react-use'
+import type { AsyncState } from 'react-use/lib/useAsyncFn'
+
+export function useTradeGasLimit(tradeComputed: TradeComputed<SwapQuoteOneResponse> | null): AsyncState<number> {
+    const { targetChainId } = TargetChainIdContext.useContainer()
+
+    const web3 = useWeb3({ chainId: targetChainId })
+    const account = useAccount()
+    const config = useMemo(() => {
+        if (!account || !tradeComputed?.trade_) return null
+        return {
+            from: account,
+            ...pick(tradeComputed.trade_, ['to', 'data', 'value']),
```

Comment:
Do not use pick. It's not type safe.

Reply:
@Jack-Works what is the safe typo in this case?

Reply:
pickBy?